x-pack/auditbeat/module/system/login/utmp_c.go

```diff
@@ -0,0 +1,120 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build linux
+
+// Pure Go reader for UTMP formatted files.
+// See utmp(5) and getutent(3) for the C structs and functions this is
+// replacing.
+
+package login
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+	"time"
+	"unsafe"
+)
+
+var byteOrder = getByteOrder()
+
+func getByteOrder() binary.ByteOrder {
+	var b [2]byte
+	*((*uint16)(unsafe.Pointer(&b[0]))) = 1
+	if b[0] == 1 {
+		return binary.LittleEndian
+	}
+	return binary.BigEndian
+}
+
+// UtType represents the ut_type field. See utmp(5).
+type UtType int16
+
+// Possible values for UtType.
+const (
+	EMPTY         UtType = 0
+	RUN_LVL              = 1
```

Comment:
I think you need to specify `UtType` for each of these. https://play.golang.org/p/Y169ZzumpVT

Reply:
done